docker-registry-sync/reposync/src/reposync/dregsy_config.py

```diff
@@ -0,0 +1,125 @@
+import yaml
+
+from typing import Dict, Tuple
+from dataclasses import dataclass, field
+from collections import deque
+
+from .prepare_stages import Stage, List
+from .utils import encode_credentials
+
+
+class BaseSerializable:
+    def to_dict(self) -> Dict:
+        raise NotImplementedError("Must implement in subclass")
+
+
+@dataclass
+class Mapping(BaseSerializable):
+    from_field: str
+    to_field: str
+    tags: List[str]
+
+    def to_dict(self) -> Dict:
+        result = {
+            "from": self.from_field,
+            "to": self.to_field,
+        }
+        if self.tags is not None and len(self.tags) > 0:
+            result["tags"] = self.tags
+        return result
+
+
+@dataclass
+class TaskRegistry(BaseSerializable):
+    registry: str
+    auth: str
+    skip_tls_verify: bool
+
+    def to_dict(self) -> Dict:
```

Comment:
Why don't  you use [asdict](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict) here?? You can use a special factory (second argument) that convert ``_`` int ``-``.

Reply:
Did you check this?